test/functional/feature_llmq_chainlocks_automatic.py

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/env python3
+# Copyright (c) 2025 The Dash Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+'''
+feature_llmq_chainlocks_automatic.py
+
+Tests automatic chainlock detection and processing from coinbase transactions.
+This test specifically validates the automatic detection feature where chainlock
+signatures embedded in coinbase transactions are automatically processed when
+blocks are connected.
+'''
+
+import time
+from test_framework.test_framework import DashTestFramework
+from test_framework.util import assert_equal, assert_raises_rpc_error, force_finish_mnsync, wait_until_helper
+
+
+class LLMQChainLocksAutomaticTest(DashTestFramework):
+    def set_test_params(self):
+        self.set_dash_test_params(6, 5)
```

Comment:
consider ab42708dac315e367ae75e9b9654bfa7b986b6be

firstly, you don't need to test masternode functionality, only just "some" chainlocks. Single-node-quorum speed up this functional test significantly.

Secondly, you should not isolate masternode for this functional test, because it adds instability in case if this masternode is in quorum. Instead, isolate regular node.

It takes only 18 seconds to run with my changes.